### DISCUSSION TITLE
Segment Profiles Events Timeline

### DISCUSSION THREAD

**Discussion by twilio-jyoung** [twilio-labs/paste#2794] [Ideas] [answered]
I've been thinking about how to best visualize various segment events to get a better picture of everything happening with a customer.  

The only component that I can find anywhere on the internet that comes close to my vision is this one: https://rawgit.com/patternfly/patternfly-timeline/master-dist/demo/

I essentially want to group types or categories of events, and display them on a timeline that you can zoom in on, and view details of specific events.  The only thing that I would change from the example above is that I would love to expand groups (accordion) to see subtypes in separate lines.  (think twilio.com as a parent, and /, /docs, /blogs, /console, etc... as children)

This would be a SUPER complex component, but one that is immensely valuable.
<img width="1214" alt="Screenshot 2022-11-01 at 12 38 39 PM" src="https://user-images.githubusercontent.com/44909531/199300761-356d1dd0-51df-4915-9a16-a8067b219f62.png">

**Comment by loreina** [comment, marked as answer]
Hi @twilio-jyoung, 

Thanks for the suggestion! This is a pretty specific visualization and could be hard to prioritize adding this to the system. We'd suggest looking into [Highcharts](https://www.highcharts.com/) to see if they offer something similar instead.
